# Function malloc_stats is only available on GLIBC, so check that before
# defining HAVE_MALLOC_STATS
# Little bit modified version of void linux patch, refer here:
# https://github.com/void-linux/void-packages/blob/master/srcpkgs/blender/patches/0001-musl-fixes.patch
--- a/intern/guardedalloc/CMakeLists.txt
+++ b/intern/guardedalloc/CMakeLists.txt
@@ -1,6 +1,12 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 # Copyright 2006 Blender Foundation. All rights reserved.

+INCLUDE(CheckSymbolExists)
+CHECK_SYMBOL_EXISTS(malloc_stats "malloc.h" HAVE_MALLOC_STATS)
+if (HAVE_MALLOC_STATS)
+	add_compile_definitions(HAVE_MALLOC_STATS)
+endif()
+
 set(INC
   .
   ../atomic
--- a/intern/guardedalloc/intern/mallocn_intern.h
+++ b/intern/guardedalloc/intern/mallocn_intern.h
@@ -17,8 +17,7 @@
 #undef HAVE_MALLOC_STATS
 #define USE_MALLOC_USABLE_SIZE /* internal, when we have malloc_usable_size() */

-#if defined(__linux__) || (defined(__FreeBSD_kernel__) && !defined(__FreeBSD__)) || \
-    defined(__GLIBC__)
+#if defined(HAVE_MALLOC_STATS) || (defined(__FreeBSD_kernel__) && !defined(__FreeBSD__))
 #  include <malloc.h>
 #  define HAVE_MALLOC_STATS
 #elif defined(__FreeBSD__)
